CMakeLists.txt

```diff
@@ -882,13 +914,61 @@ else()
                       INCLUDES
                       ${draco_include_paths}
                       OBJLIB_DEPS
-                      draco_unity_plugin
+                      ${unity_objlib_dependency}
+                      LIB_DEPS
+                      ${unity_lib_dependency}
+                      )
+
+    draco_add_library(NAME
+                      dracoenc_unity
+                      TYPE
+                      ${unity_decoder_lib_type}
+                      DEFINES
+                      ${draco_defines}
+                      INCLUDES
+                      ${draco_include_paths}
+                      OBJLIB_DEPS
+                      draco_unity_enc_plugin
                       LIB_DEPS
                       ${draco_plugin_dependency})
 
     # For Mac, we need to build a .bundle for the unity plugin.
     if(APPLE)
       set_target_properties(dracodec_unity PROPERTIES BUNDLE true)
+      set_target_properties(dracoenc_unity PROPERTIES BUNDLE true)
+    endif()
+    if(IOS)
+      # Workaround to ensure bitcode is generated when
+      # generating an iOS library with Makefiles as generator.
+      target_compile_options( dracodec_unity PUBLIC -fembed-bitcode )
+      target_compile_options( draco_attributes PUBLIC -fembed-bitcode )
+      target_compile_options( draco_compression_attributes_dec PUBLIC -fembed-bitcode )
+      target_compile_options( draco_compression_bit_coders PUBLIC -fembed-bitcode )
+      target_compile_options( draco_compression_decode PUBLIC -fembed-bitcode )
+      target_compile_options( draco_compression_entropy PUBLIC -fembed-bitcode )
+      target_compile_options( draco_compression_mesh_dec PUBLIC -fembed-bitcode )
+      target_compile_options( draco_compression_point_cloud_dec PUBLIC -fembed-bitcode )
+      target_compile_options( draco_core PUBLIC -fembed-bitcode )
+      target_compile_options( draco_dec_config PUBLIC -fembed-bitcode )
+      target_compile_options( draco_io PUBLIC -fembed-bitcode )
+      target_compile_options( draco_mesh PUBLIC -fembed-bitcode )
+      target_compile_options( draco_metadata PUBLIC -fembed-bitcode )
+      target_compile_options( draco_metadata_dec PUBLIC -fembed-bitcode )
+      target_compile_options( draco_point_cloud PUBLIC -fembed-bitcode )
+      target_compile_options( draco_points_dec PUBLIC -fembed-bitcode )
+      target_compile_options( draco_unity_plugin PUBLIC -fembed-bitcode )
+    endif()
+    if (WIN32 AND ${CMAKE_SYSTEM_NAME} STREQUAL "WindowsStore")
```

Comment:
Same with this one, but we already have MSVC specific flags. Just update the draco_msvc_cxx_flags list, and add /wd4146.